### DISCUSSION TITLE
Badge color in hex

### DISCUSSION THREAD

**Discussion by labs20** [quasarframework/quasar#14739] [General - Components / Directives / etc]
Hi. Looks like I can't set the q-badge color property to an hexadecimal value, it only accepts quasar pallete's color name.

Is that intended? 

[https://codepen.io/labs20/pen/abGYpob](https://codepen.io/labs20/pen/abGYpob)

Thanks

**Comment by pdanpdan** [comment]
Yes. If you want you can define your color as ,bg-xxx / .text-xxx and use xxx as color

**Reply by labs20** [reply]
Hi. Thanks for your quick answer! =]

I have a problem with that, because I need to let my users to choose a custom color that will be applied to the badge.

Why the color prop of the q-badge behaves different from other comps, like toggle for example?

Thanks!

**Comment by metalsadman** [comment]
make your own brand like what pdan suggested, assign a css variable to it that then you can change dynamically using the color utils https://codepen.io/metalsadman/pen/OJvKbov

**Reply by labs20** [reply]
Hi. Thanks for the pen, I'll make good use of it for sure.

But still, seems to me like a bug that `color` on q-badge has different behaviour than `color` on other comps, and while the workaround sure will resolve the problem, I would prefer no workaround at all and just do `color="#ff76ff"`.

Thanks for your help!

**Reply by metalsadman** [reply]
I don't think not many color props accepts hex at all other than brand or using purely css.

**Reply by rstoenescu** [reply]
There would be significant overhead and framework footprint (in kb) to allow color props to work directly with hex values.

**Comment by Araz0** [comment]
i know this is an old discussion, but it was the best result i got when i searched so to help others with the same issue, my workaround was to set class prop on the component and do this on the component. essentially just overriding its own color variable.
`class="[--q-primary:theme(colors.ni-green-pale)]"` (for tailwind for example)
or
`class="[--q-primary: #FFFFFF]"` (for normal hex)

hope it helps someone like me in the future haha
